### PR TITLE
chore: make fetch remote host timeout configurable for test

### PR DESCRIPTION
**Pull request by nateab**
### Description 
_What behavior do you want to change, why, how does your patch achieve the changes?_

`TerminateTransientQueryFunctionalTest` has been historically flaky, and the most recent fix was to improve the logging https://github.com/confluentinc/ksql/pull/9784.
`[2023-04-19 09:42:55,508] WARN Failed to retrieve info from host: HostInfo{host='localhost', port=1621}, statement: terminate transient_PAGEVIEW_KSTREAM_8865619672769769280;, cause: {} (io.confluent.ksql.rest.server.execution.RemoteHostExecutor:135) java.util.concurrent.TimeoutException
`
This new improved logging indicated the issue was with the timeout in RemoteHostExecutor, which has it configured to 10 seconds, so we bump this up 20 seconds in this PR.

Address feedback on https://github.com/confluentinc/ksql/pull/9890
### Testing done 
_Describe the testing strategy. Unit and integration tests are expected for any behavior changes._

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")
- [ ] Do these changes have compatibility implications for rollback? If so, ensure that the ksql [command version](https://github.com/confluentinc/ksql/blob/master/ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/Command.java#L41) is bumped.

### CODE REVIEW

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/RemoteHostExecutor.java [hunk 1]
```diff
@@ -143,4 +145,15 @@ public Pair<Map<HostInfo, KsqlEntity>, Set<HostInfo>> fetchAllRemoteResults() {
...
     }
   }
+
+  private long getFetchRemoteHostsTimeout(){
+    final Object timeout
+        = executionContext.getKsqlConfig().getLong(KSQL_FETCH_REMOTE_HOSTS_TIMEOUT_SECONDS);
+
+    if (timeout instanceof Long) {
+      return (long) timeout;
+    }
```
Comment (lihaosky) [inline ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/RemoteHostExecutor.java:155]: nit: I think `getLong` already returns `Long`?

**Review by lihaosky — APPROVED** [review]
Thanks for fixing!

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/RemoteHostExecutor.java (hunk 1 above)
Comment (nateab) [reply ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/RemoteHostExecutor.java:155]: yeah but if the config is not set it will return null, which is why we check if it is indeed `Long`

Comment (lihaosky) [reply ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/RemoteHostExecutor.java:155]: Will it use the default you defined if not set? `KSQL_FETCH_REMOTE_HOSTS_TIMEOUT_SECONDS_DEFAULT`

Comment (nateab) [reply ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/RemoteHostExecutor.java:155]: I did not include it in the first commit of this pr and i got a bunch of test failures due to null exceptions when the config was not set like this 
https://jenkins.confluent.io/job/Confluent%20Public%20Repo%20PR%20builder/job/ksql/job/PR-9894/1/testReport/junit/io.confluent.ksql.rest.server.execution/ListQueriesExecutorTest/shouldListQueriesBasic/

I followed the pattern here for how to get the config/use the default if not set https://github.com/confluentinc/ksql/blob/19a8fcc67bdf582bd9bebe9778197e8238daf389/ksqldb-cli/src/main/java/io/confluent/ksql/cli/Cli.java#L455-L464

Comment (lihaosky) [reply ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/RemoteHostExecutor.java:155]: That's weird. I saw `getLong` used like this: 

https://github.com/confluentinc/ksql/blob/master/ksqldb-engine/src/main/java/io/confluent/ksql/security/KsqlCacheAccessValidator.java#L118-L119

Comment (nateab) [reply ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/RemoteHostExecutor.java]: Nvm you are right, the tests were just broken, i removed the `getFetchRemoteHostsTimeout` function and fixed the tests to avoid the null pointer exceptions, good catch thank you!
